hack/generator/pkg/test/fake_function.go

```diff
@@ -13,12 +13,14 @@ import (
 )
 
 // FakeFunction is a fake function that can be used for testing purposes
+// Note that there's another FakeFunction in astmodel but we can't eliminate that one because it would cause a package
+// reference cycle
 type FakeFunction struct {
-	name       string
-	Imported   *astmodel.PackageReferenceSet
-	Referenced astmodel.TypeNameSet
-	returnType astmodel.Type
-	idFactory  astmodel.IdentifierFactory
+	name         string
+	Imported     *astmodel.PackageReferenceSet
+	Referenced   astmodel.TypeNameSet
+	TypeReturned astmodel.Type
```

Comment:
super minor: I have a slight preference for ReturnType here. 
Oh, I see that you did this to make it consistent with Imported and References. Present tense (Imports, References) feels more natural for those too, and I quite like that they can also be read as plural nouns as well since they're sets. 🤷

Reply:
Also, the function `ReturnType()` is on the `Function` interface, so I couldn't use `ReturnType` for the member name here.